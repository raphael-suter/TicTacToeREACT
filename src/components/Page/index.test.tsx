import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import App from '.';

const setup = () => shallow(<App />);

describe('Page', () => {
  test('Renders correctly.', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Shows Dialogs.', () => {
    const wrapper = setup();

    wrapper.setState({
      userDialogVisible: false,
      messageDialogVisible: false,
      loaderVisible: false
    });

    const childrenCountBefore = wrapper.children().length;

    wrapper.setState({
      userDialogVisible: true,
      messageDialogVisible: true,
      loaderVisible: true
    });

    const childrenCountAfter = wrapper.children().length;
    expect(childrenCountAfter - childrenCountBefore).toBe(3);
  });
});

describe('Page, verifyPlayers', () => {
  const wrapper = setup();
  // TODO
});
