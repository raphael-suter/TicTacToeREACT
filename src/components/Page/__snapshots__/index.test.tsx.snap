// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Renders correctly. 1`] = `
<ThemeProvider
  theme={
    Object {
      "darkColor": "#000000",
      "lightColor": "#ffffff",
      "mainColor": "#000080",
      "mediumColor": "#f5f5f5",
      "shadowColor": "rgba(0, 0, 0, 0.3)",
    }
  }
>
  <Memo(GlobalStyleComponent) />
  <Component
    score="X 0:0 O"
    title="TicTacToe"
  >
    <Component
      icon="settings_brightness"
      onClick={[Function]}
    />
    <Component
      icon="delete"
      onClick={[Function]}
    />
  </Component>
  <Component>
    <Component
      content=""
      key="0"
      onClick={[Function]}
    />
    <Component
      content=""
      key="1"
      onClick={[Function]}
    />
    <Component
      content=""
      key="2"
      onClick={[Function]}
    />
    <Component
      content=""
      key="3"
      onClick={[Function]}
    />
    <Component
      content=""
      key="4"
      onClick={[Function]}
    />
    <Component
      content=""
      key="5"
      onClick={[Function]}
    />
    <Component
      content=""
      key="6"
      onClick={[Function]}
    />
    <Component
      content=""
      key="7"
      onClick={[Function]}
    />
    <Component
      content=""
      key="8"
      onClick={[Function]}
    />
  </Component>
  <Component>
    <Component />
  </Component>
</ThemeProvider>
`;
